cmd/statusd/main.go

```diff
@@ -110,6 +122,36 @@ func startDebug(backend *api.StatusBackend) error {
 	return err
 }
 
+// startCollectingStats collects various stats about the node and other protocols like Whisper.
+func startCollectingStats(interruptCh <-chan struct{}, nodeManager common.NodeManager) {
+	log.Printf("Starting stats on %v", *statsAddr)
+
+	node, err := nodeManager.Node()
+	if err != nil {
+		log.Printf("Failed to run metrics because could not get node: %v", err)
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		if err := nodeMetrics.SubscribeServerEvents(ctx, node); err != nil {
+			log.Printf("Failed to subscribe server events: %v", err)
+		}
+	}()
+
+	server := metrics.NewMetricsServer(*statsAddr)
+	go func() {
+		log.Printf("Metrics server failed: %v", server.ListenAndServe())
```

Comment:
I'd extracted `err := server.ListenAndServe()` to the previous line for readability purposes.

Reply:
also we probably should ignore `http.ErrServerClosed` if returned from `ListenAndServe`.

> After Shutdown or Close, the returned error is ErrServerClosed.

https://golang.org/pkg/net/http/#Server.Serve